JaCoCo reports not working

At some point (perhaps with a Gradle update?) the JaCoCo reports stopped working. For example:

https://github.com/micronaut-projects/micronaut-openapi/actions/runs/5329318280/jobs/9654878870?pr=1059

```
No coverage report can be found with sonar.coverage.jacoco.xmlReportPaths='/home/runner/work/micronaut-openapi/micronaut-openapi/build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml'. Using default locations: target/site/jacoco/jacoco.xml,target/site/jacoco-it/jacoco.xml,build/reports/jacoco/test/jacocoTestReport.xml
```

The aggregator project certainly doesn't contain the report. But neither the individual projects. If I try to run `jacocoTestReport` directly, I get:

```
Exception in thread "OkHttp TaskRunner" java.lang.NoClassDefFoundError: kotlin/jvm/internal/Ref$IntRef
        at okhttp3.internal.http2.Http2Connection.shutdown(Http2Connection.kt:419)
        at okhttp3.internal.http2.Http2Connection.close$okhttp(Http2Connection.kt:449)
        at okhttp3.internal.http2.Http2Connection$ReaderRunnable.invoke(Http2Connection.kt:627)
        at okhttp3.internal.http2.Http2Connection$ReaderRunnable.invoke(Http2Connection.kt:609)
        at okhttp3.internal.concurrent.TaskQueue$execute$1.runOnce(TaskQueue.kt:98)
        at okhttp3.internal.concurrent.TaskRunner.runTask(TaskRunner.kt:116)
        at okhttp3.internal.concurrent.TaskRunner.access$runTask(TaskRunner.kt:42)
        at okhttp3.internal.concurrent.TaskRunner$runnable$1.run(TaskRunner.kt:65)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ClassNotFoundException: kotlin.jvm.internal.Ref$IntRef
        ... 11 more
```

Build is successful, but no report is created.

--- a/micronaut-gradle-plugins/src/functionalTest/groovy/io/micronaut/build/quality/QualityReportingPluginFunctionalTest.groovy
+++ b/micronaut-gradle-plugins/src/functionalTest/groovy/io/micronaut/build/quality/QualityReportingPluginFunctionalTest.groovy
@@ -21,6 +21,24 @@ class QualityReportingPluginFunctionalTest extends AbstractFunctionalTest {
         file("build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml").exists()
     }
 
+    void "jacocoTestReport runs aggregate coverage reports"() {
+        given:
+        withSample("test-micronaut-module")
+        file("gradle.properties") << "micronaut.jacoco.enabled=true"
+
+        when:
+        run 'jacocoTestReport'
+
+        then:
+        tasks {
+            succeeded ':subproject1:test'
+            succeeded ':subproject2:test'
+            succeeded ':testCodeCoverageReport'
+            succeeded ':jacocoTestReport'
+        }
+        file("build/reports/jacoco/testCodeCoverageReport/testCodeCoverageReport.xml").exists()
+    }
+
     void "it can run Spotless and Checkstyle"() {
         given:
         withSample("test-micronaut-module")

--- a/micronaut-gradle-plugins/src/main/java/io/micronaut/build/MicronautQualityReportingAggregatorPlugin.java
+++ b/micronaut-gradle-plugins/src/main/java/io/micronaut/build/MicronautQualityReportingAggregatorPlugin.java
@@ -81,9 +81,10 @@ public class MicronautQualityReportingAggregatorPlugin implements Plugin<Project
         rootProject.afterEvaluate(unused -> {
             if (!rootProject.getTasks().getNames().contains("jacocoTestReport")) {
                 rootProject.getTasks().register("jacocoTestReport", t ->  {
-                   t.setDescription("Placeholder task for Jacoco reports");
+                   t.setDescription("Generates aggregate Jacoco reports");
+                   t.dependsOn(COVERAGE_REPORT_TASK_NAME);
                    t.doFirst(unused2 -> {
-                       System.out.println("Placeholder task for Jacoco reports");
+                       System.out.println("Generating aggregate Jacoco reports");
                    });
                 });
             }

